File: gunhoflash/find-minimum-in-rotated-sorted-array.ts
function findMin(nums: number[]): number {
  let leftIndex = 0;
  let rightIndex = nums.length - 1;

  while (leftIndex < rightIndex) {
    const midIndex = Math.floor((leftIndex + rightIndex) / 2);
    const midValue = nums[midIndex];
    const rightValue = nums[rightIndex];
  
    if (midValue < rightValue) {
      rightIndex = midIndex;
    } else {
      leftIndex = midIndex + 1;
    }
  }

  return nums[leftIndex];
}

File: gunhoflash/find-minimum-in-rotated-sorted-array.py
from typing import List

class Solution:
  def findMin(self, nums: List[int]) -> int:
    leftIndex = 0
    rightIndex = len(nums) - 1

    while leftIndex < rightIndex:
      midIndex = (leftIndex + rightIndex) // 2
      midValue = nums[midIndex]
      rightValue = nums[rightIndex]

      if midValue < rightValue:
        rightIndex = midIndex
      else:
        leftIndex = midIndex + 1

    return nums[leftIndex]
